### PR TITLE
Add back HttpRequest.LogonUserIdentity

### LINKED ISSUE #525: Add back `HttpRequest.LogonUserIdentity`
## Summary

`HttpRequest.LogonUserIdentity` was removed few days ago in #510. But it is being actively used in our codebase. Please add it back.

## Motivation and goals

1-2 paragraphs, or a bullet-pointed list. What existing pain points does this solve? What evidence shows it's valuable to solve this?

## In scope

A list of major scenarios, perhaps in priority order.

## Out of scope

Scenarios you explicitly want to exclude.

## Risks / unknowns

How might developers misinterpret/misuse this? How might implementing it restrict us from other enhancements in the future? Also list any perf/security/correctness concerns.

## Examples

```c#
switch (context.Request.LogonUserIdentity.AuthenticationType.ToUpper(CultureInfo.InvariantCulture))
{
......
}
```

```c#
IntPtr userToken = context.Request.LogonUserIdentity.Token
```

### PR DESCRIPTION
**Pull request by twsouthwick**
Previously, this had been added typed as IIdentity which was incorrect. This adds it back as WindowsIdentity and exposes it via a feature so it can be easily customized if needed. By default, it will use the current user and check if it has a WindowsIdentity and return that.

Fixes #525

### CODE REVIEW

**Comment by twsouthwick** [comment]
@joperezr can I get a review on this?

File: src/Microsoft.AspNetCore.SystemWebAdapters/Microsoft.AspNetCore.SystemWebAdapters.csproj [hunk 1]
```diff
@@ -31,6 +31,8 @@
     <Compile Include="Adapters/SessionState/ISessionState.cs" />
     <Compile Include="IHtmlString.cs" />
     <Compile Include="HtmlString.cs" />
+
+    <PackageReference Include="System.Security.Principal.Windows" Version="4.3.0" />
```
Comment (joperezr) [inline src/Microsoft.AspNetCore.SystemWebAdapters/Microsoft.AspNetCore.SystemWebAdapters.csproj:35]: Why this version as opposed to a newer one (5.0.0 or 4.7.0)?

**Review by joperezr — APPROVED** [review]

**Comment by joperezr** [comment]
Left a comment but looks good otherwise.

File: src/Microsoft.AspNetCore.SystemWebAdapters/Microsoft.AspNetCore.SystemWebAdapters.csproj (hunk 1 above)
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters/Microsoft.AspNetCore.SystemWebAdapters.csproj:35]: not sure - the latest VS found was 4.3.0 - I"ll change it to 5.0.0

**Pull request merged**